sw/device/tests/sim_dv/flash_rma_unlocked_test.c

```diff
@@ -0,0 +1,298 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_lc_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/pinmux.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+#define LC_TOKEN_SIZE 16
+
+static dif_uart_t uart0;
+static dif_flash_ctrl_state_t flash_state;
+
+enum {
+  kFlashInfoPageIdCreatorSecret = 1,
+  kFlashInfoPageIdOwnerSecret = 2,
+  kFlashInfoPageIdIsoPart = 3,
+  kFlashInfoBank = 0,
+  kRegionBaseBank0Page0Index = 0,
+  kRegionBaseBank0Page255Index = 255,
+  kRegionBaseBank1Page0Index = 256,
+  kRegionBaseBank1Page255Index = 511,
+  kFlashBank0Page0DataRegion = 0,
+  kFlashBank0Page255DataRegion = 1,
+  kFlashBank1Page0DataRegion = 2,
+  kFlashBank1Page255DataRegion = 3,
+  kPartitionId = 0,
+  kRegionSize = 1,
+  kDataSize = 16,
+  kPageSize = 2048,
+};
+
+const uint32_t kRandomData1[kDataSize] = {
```

Comment:
These should all be static... also, would it make sense for this to be a `kRandomData[kDataSize][8]`?